pkg/config/zonepb/zone.go

```diff
@@ -1094,6 +1094,75 @@ func (z *ZoneConfig) LeasePreference(i int) cat.ConstraintSet {
 	return &z.LeasePreferences[i]
 }
 
+func (z *ZoneConfig) ToSpanConfig() (roachpb.SpanConfig, error) {
+	var sc roachpb.SpanConfig
+	// Copy over the values.
+	sc.RangeMinBytes = *z.RangeMinBytes
+	sc.RangeMaxBytes = *z.RangeMaxBytes
+	sc.GCTTL = z.GC.TTLSeconds
+
+	// GlobalReads is false by default.
+	sc.GlobalReads = false
+	if z.GlobalReads != nil {
+		sc.GlobalReads = *z.GlobalReads
+	}
+	sc.NumReplicas = *z.NumReplicas
+
+	// NumVoters = NumReplicas if NumVoters is unset.
+	sc.NumVoters = *z.NumReplicas
+	if z.NumVoters != nil {
+		sc.NumVoters = *z.NumVoters
+	}
+
+	translateConstraints := func(src []Constraint, dest []roachpb.Constraint) error {
```

Comment:
Don't take a destination slice, return a []roachpb.Constraint instead. Name this helper to toSpanConfigConstraints.